src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +110,10 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter((Predicate<? super Candidate>) new CandidateValidator())
```

Comment:
```suggestion
                .filter(new CandidateValidator())
```
Better make it class-level variable